tests/test_ethash.py

```diff
@@ -363,12 +348,19 @@ def test_dataset_generation_random_epoch(tmpdir: str) -> None:
         2. Randomly take 500 indices between
         [101, `dataset size in words` - 1] and ensure that the values are
         same between python implementation and DAG dataset.
+
+    NOTE - For this test case to run, it is mandatory for Geth to be
+    installed and accessible
     """
-    download_geth(tmpdir)
+    geth_path = shutil.which("geth")
+    if geth_path == None:
```

Comment:
Testing for `None` should be done with `is`:

```suggestion
    if geth_path is None:
```